pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -203,8 +207,14 @@ func ConvertIngressVirtualService(ingress knetworking.Ingress, domainSuffix stri
 		// see https://kubernetes.io/docs/concepts/services-networking/ingress/#multiple-matches
 		vs := ingressByHost[host].Spec.(*networking.VirtualService)
 		sort.SliceStable(vs.Http, func(i, j int) bool {
-			r1Len, r1Ex := getMatchURILength(vs.Http[i].Match[0])
-			r2Len, r2Ex := getMatchURILength(vs.Http[j].Match[0])
+			var r1Len, r2Len int
+			var r1Ex, r2Ex bool
+			if vs.Http[i].Match != nil || len(vs.Http[i].Match) != 0 {
```

Comment:
nit: it might be simpler to just make the `nil` ones `Prefix: /` to make things simpler. I think it will end up the same in Envoy in the end, so just a matter of which is simpler in the code

Reply:
Good point, that would work too! My only hesitation with that approach is that requires all URLs to have the trailing slash. In the case where there is no slash, I believe `Prefix: /` would fail. If /* means all paths including empty paths then I am not sure `Prefix: /` is sufficient. Is `Prefix: ` allowed?

Reply:
`prefix: /` is the end result in envoy config for all configs - even a plain HTTP service without a route. it will match all traffic. `http://example.com` is treated as `path=/` (not sure if that is an Envoy thing or some HTTP RFC, but that is how it ends up working in the end)